### PR TITLE
Add option for specifying stored/virtual on computed columns

### LINKED ISSUE #6682: HasComputedColumnSql missing option for persisting computed values
### The issue

I am using the HasComputedColumnSql extension which basically works for me as expected. 

However, I am missing an option to instruct EF to persist the computed values. 

Currently the code that is generated for my computed column is:

``` SQL
[UrlString] AS [Schema] + ':/' + CASE WHEN [DomainPrefix] IS NULL THEN '' ELSE '/' + [DomainPrefix] + '.' END + CASE WHEN [DomainPrefix] IS NULL THEN '/' ELSE '' END + [ApplicationDomain] + CASE WHEN [InternalPath] IS NULL THEN '' ELSE '/' + [InternalPath] END 
```

However, using an option like .Persisted() or a boolean flag like HasComputedColumnSql(string sql, bool persist) I would like to see code generated like this:

``` SQL
[UrlString] AS [Schema] + ':/' + CASE WHEN [DomainPrefix] IS NULL THEN '' ELSE '/' + [DomainPrefix] + '.' END + CASE WHEN [DomainPrefix] IS NULL THEN '/' ELSE '' END + [ApplicationDomain] + CASE WHEN [InternalPath] IS NULL THEN '' ELSE '/' + [InternalPath] END PERSISTED
```

It would be an extremely useful feature as it would save time in scripting it manually.
### Further technical details

EF Core version: 1.0.0
Operating system: Windows 10
Visual Studio version: VS 2015 Update 3

### PR DESCRIPTION
**Pull request by roji**
Closes #6682

# Cross-database situation

| Database   | Stored name | Virtual name                  | Default                                  | Docs |
|------------|-------------|-------------------------------|------------------------------------------|------|
| SQL Server | PERSISTED   | &lt;empty&gt;                 | &lt;empty&gt; (virtual)                  | https://docs.microsoft.com/en-us/sql/relational-databases/tables/ |specify-computed-columns-in-a-table?view=sql-server-ver15
| Sqlite     | STORED      | VIRTUAL                       | VIRTUAL                                  | https://www.sqlite.org/gencol.html |
| PostgreSQL | STORED      | VIRTUAL (not yet implemented) | None, STORED must currently be specified | https://www.postgresql.org/docs/12/ddl-generated-columns.html |
| MySQL      | STORED      | VIRTUAL                       | VIRTUAL                                  | https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html |

# Notes

* tl;dr everyone calls it STORED except for SQL Server (surprise) which calls it PERSISTED. So I chose STORED as our naming.
* This currently introduces a new `ComputedColumnIsStored` annotation alongside `ComputedColumnSql`. @AndriySvyryd I know you prefer to merge annotations where possible for perf, but I'm hesitant to encode both SQL and other info in the same annotation string. We can change the annotation format to `<IsStored>,<SQL>` and identify old annotations not starting with true/false and a comma, but since SQL is arbitrary SQL it seems a bit dangerous.
* If we keep two annotations, what should be the behavior of HasComputedColumnSql accepting IConventionPropertyBuilder? Should we have two separate ones - one for the SQL annotation and another for the IsStored - or one which checks both and only updates if both are settable? Or each one if it's settable?
* Note that this introduces a binary break by adding a default argument to HasComputedColumnSql (but recompiling is OK). If we want to avoid this we can add an overload instead.
* Instead of a `stored` bool we could have a ComputedColumnType enum (Stored/Virtual), but this seems needless verbose/heavy.

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
For IConventionPropertyBuilder we should have two separate methods

**Comment by roji** [comment]
> For IConventionPropertyBuilder we should have two separate methods

Thanks, done (HasComputedColumnIsStored is somewhat weird naming but consistent etc.)

**Comment by AndriySvyryd** [comment]
IsStoredComputedColumn would be better IMO

**Comment by roji** [comment]
Rebased on latest master (resolving conflicts with #20602), squashed and renamed HasComputedColumnIsStored->IsStoredComputedColumn

File: src/EFCore.Relational/Extensions/RelationalPropertyBuilderExtensions.cs [hunk 1]
```diff
@@ -360,15 +361,22 @@ public static class RelationalPropertyBuilderExtensions
...
             [NotNull] this PropertyBuilder propertyBuilder,
-            [CanBeNull] string sql)
+            [CanBeNull] string sql,
+            bool? stored = null)
         {
             Check.NotNull(propertyBuilder, nameof(propertyBuilder));
             Check.NullButNotEmpty(sql, nameof(sql));
 
             propertyBuilder.Metadata.SetComputedColumnSql(sql);
+            propertyBuilder.Metadata.SetComputedColumnIsStored(stored);
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Extensions/RelationalPropertyBuilderExtensions.cs:379]: Only do this if `stored` is not null

**Comment by roji** [comment]
Thanks @AndriySvyryd!
